contracts/SecurityTokenRegistry.sol

```diff
@@ -227,20 +228,18 @@ contract SecurityTokenRegistry is ISecurityTokenRegistry, EternalStorage {
      * @notice Internal -- Modifies the ticker details.
      */
     function _modifyTicker(address _owner, string _ticker, string _tokenName, uint256 _registrationDate, uint256 _expiryDate, bool _status) internal {
-        address currentOwner = getAddress(Encoder.getKey("registeredTickers_owner", _ticker));
-        if (currentOwner == address(0) && _registrationDate == 0 && _expiryDate == 0) {
-            _addTicker(_owner, _ticker, _tokenName, now, now.add(getUint(Encoder.getKey("expiryLimit"))), _status, true);
-            return;
+        address currentOwner = _tickerOwner(_ticker);
+        if (currentOwner != address(0)) {
```

Comment:
@satyamakgec 

I changed the logic so that if you're creating a ticker through a call to `modifySecurityToken` then the registration / expiry times are set as `now` and `now + expiryLimit`.